frontend/src/metabase/search/components/SearchResult/components/CollectionIcon.tsx

```diff
@@ -0,0 +1,16 @@
+import { Icon } from "metabase/core/components/Icon";
+import { PLUGIN_COLLECTIONS } from "metabase/plugins";
+import { DEFAULT_ICON_SIZE } from "metabase/search/components/SearchResult/components";
+import type { WrappedResult } from "metabase/search/types";
+
+export function CollectionIcon({ item }: { item: WrappedResult }) {
+  const iconProps = { ...item.getIcon() };
+  const isRegular = PLUGIN_COLLECTIONS.isRegularCollection(item.collection);
+  if (isRegular) {
+    iconProps.size = DEFAULT_ICON_SIZE;
+  } else {
+    iconProps.width = 20;
+    iconProps.height = 24;
+  }
+  return <Icon {...iconProps} tooltip={null} />;
```

Comment:
I know this code is just moved, but it does look confusing.

```suggestion
  const iconProps = { ...item.getIcon(), tooltip: null };
  const isRegular = PLUGIN_COLLECTIONS.isRegularCollection(item.collection);

  if (isRegular) {
    return <Icon {...iconProps} size={DEFAULT_ICON_SIZE} />;
  }

  return <Icon {...iconProps} width={20} height={24} />;
```